src/lexer.rs

```diff
@@ -288,4 +390,158 @@ impl JavascriptLexer {
+    let mut pending_export = self.pending_export.as_mut().unwrap();
+
+    match pending_export.expected_token {
+      ExportToken::Variables => {
+        match self.current_char {
+          '{' => {
+            pending_export.expected_token = ExportToken::NamedExport;
+          },
+          '*' => {
+            pending_export.expected_token = ExportToken::From;
+          },
+          _ => {}
+        }
+
+        self.keep_using_handler();
+      },
+      ExportToken::NamedExport => {
+        if pending_export.token_start.is_none() {
+          match self.current_char {
+            c if c.is_whitespace() => {},
+            '\u{0041}'..='\u{2FA1D}' => {
+              pending_export.token_start = Some(self.current_index);
+            },
+            _ => {
+              panic!(format!("Invalid character '{}' at index {} - expected identifier start", self.current_char, self.current_index));
+            }
+          }
+        } else {
+          match self.current_char {
+            '\u{0030}'..='\u{E01EF}' => {},
+            _ => {
+              let identifier = String::from(self.source.get(pending_export.token_start.unwrap()..self.current_index).unwrap());
+              if identifier == "default" {
+                  // TODO: this can be improved.  "default as" w/ backtracking?
+                  // reset since next identifier will alias default "export { default as ... }"
+                  pending_export.token_start = None;
+                  pending_export.expected_token = ExportToken::NextNamedExport;
+              } else {
+                  let named_export = NamedExport {
+                    variable_name: identifier.clone(),
+                    binding_name: identifier,
+                  };
+                  pending_export.export.named_exports.push(named_export);
+                  let next_token;
+                  match self.current_char {
+                    '}' => {
+                      next_token = ExportToken::From;
+                    },
+                    ',' => {
+                      next_token = ExportToken::NamedExport;
+                    },
+                    _ => {
+                      next_token = ExportToken::NextNamedExport;
+                    }
+                  }
+                  pending_export.expected_token = next_token;
+                  pending_export.token_start = None;
+              }
+            }
+          }
+        }
+        self.keep_using_handler();
+      },
+      ExportToken::NextNamedExport => {
+        match self.current_char {
+          '}' => {
+            pending_export.expected_token = ExportToken::From;
+          },
+          ',' => {
+            pending_export.expected_token = ExportToken::NamedExport;
+          },
+          'a' => {
+            // re-export: `export { default as b } from './b';`
+            self.indices_to_skip = 1;
+            pending_export.expected_token = ExportToken::NamedExport;
```

Comment:
Sending to ExportToken::NamedExport seems wrong to me, here. Won't this create two named exports instead of one export whose variable name is different than its binding name?

Reply:
https://github.com/rawsemble/rawsemble/pull/1/files#diff-5d5fd55cd63fa75e4c8902fa6b6d63da3f8da03742454636a8d9c092f6f6eeb4R426

So I'm not happy about this in it's current state but if the NamedExport is `default`, then I reset and continue finding a named export.  I certainly think something here can be reworked!  What do you think?

Reply:
```js
export { default as b } from './b';
```

I think the above would best be represented as a named export whose binding_name is `default`?

We can address this in a future PR - I agree with you that getting tests in place is necessary before trying to tackle all these things.